### PR TITLE
elegant naming, package structure

### LINKED ISSUE #9: MessageGenerator.java:31-32: Bad naming We got -or suffix...
The puzzle `4-3e287310` from #4 has to be resolved: 

https://github.com/Blamer-io/bot/blob/c78df2b3dd3ae9256c7e5cb2e3805bfcb956ffaa/src/main/java/io/blamer/bot/answer/generator/MessageGenerator.java#L31-L32

The puzzle was created by @ivan on 27-Apr-23. 

 role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by h1alexbel**
closes #9

---

## PR-Codex overview
This PR introduces changes to the project's conversation handling logic. It replaces the MessageGenerator interface with the Conversation interface, and updates related classes accordingly.

### Detailed summary
- Replaced MessageGenerator interface with Conversation interface
- Updated related classes to use Conversation interface
- Renamed several classes to reflect updated functionality
- Updated tests to use updated class names and interfaces

> The following files were skipped due to too many changes: `src/main/java/io/blamer/bot/answer/generator/impl/RegistryMessageGenerator.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by h1alexbel** [comment]
@l3r8yJ take a look, please

File: src/main/java/io/blamer/bot/conversation/Conversation.java
Comment (l3r8yJ) [inline src/main/java/io/blamer/bot/conversation/Conversation.java:1]: I think that the interface must be located at a level higher than its implementation, you need to create another package, such as `commands`, or raise `Conversation.java` by 1 package

wdyt

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments, please

File: src/main/java/io/blamer/bot/conversation/Conversation.java
Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/conversation/Conversation.java:1]: @h1alexbel maybe `impl` not the best name, but i think this structure will be bit better

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/Conversation.java:1]: @l3r8yJ

Comment (h1alexbel) [reply src/main/java/io/blamer/bot/conversation/Conversation.java:1]: @l3r8yJ routed

Comment (l3r8yJ) [reply src/main/java/io/blamer/bot/conversation/Conversation.java:1]: nice

**Review by l3r8yJ — APPROVED** [review]
lgtm

**Comment by l3r8yJ** [comment]
@rultor merge

**Comment by l3r8yJ** [comment]
@h1alexbel thanks!

**Pull request merged**